### PR TITLE
feat(ArgoCD): add ArgoCD Application Health and Sync Status monitor

### PR DESCRIPTION
**Pull request by tonymbugua-outreach**
## What this PR does / why we need it
We want to add basic monitoring for argocd applications and trigger when the application health of sync status is unhealthy

This template was tested in the following [PR](https://github.com/getoutreach/deploytestservice/pull/78) and produced the following monitors:
- https://app.datadoghq.com/monitors/106302048
- https://app.datadoghq.com/monitors/106302047

## Jira ID

[DT-3278]

## Notes for your reviewers

[DT-3278]: https://outreach-io.atlassian.net/browse/DT-3278?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by artemchernyak-outreach — APPROVED** [review]

File: templates/.snapshots/TestDatadogTf-monitoring-datadog.tf.tpl-monitoring-datadog.tf.snapshot [hunk 1]
```diff
@@ -38,6 +38,32 @@ locals {
...
+  require_full_window = false
+}
+
+resource "datadog_monitor" "argocd_application_sync_status" {
+  type = "query alert"
+  name = "Testing argocd application sync status"
+  query = "max(last_15m):clamp_max(sum:argocd_application_controller.argocd_app_info{name:testing,sync_status:synced} by {cluster_name}, 1) < 1"
+  tags = local.ddTags
+  message = <<EOF
+  If we ever have a pod restart, we want to know.
```
Comment (jkinkead) [inline templates/.snapshots/TestDatadogTf-monitoring-datadog.tf.tpl-monitoring-datadog.tf.snapshot:60]: This seems wrong?

File: templates/.snapshots/TestDatadogTf-monitoring-datadog.tf.tpl-monitoring-datadog.tf.snapshot [hunk 2]
```diff
@@ -38,6 +38,32 @@ locals {
   ddTags = concat(["testing", "team:test:team"], var.additional_dd_tags)
 }
 
+resource "datadog_monitor" "argocd_application_health_status" {
+  type = "query alert"
+  name = "Testing argocd application health status"
+  query = "max(last_15m):clamp_max(sum:argocd_application_controller.argocd_app_info{name:testing,health_status:healthy} by {cluster_name}, 1) < 1"
```
Comment (jkinkead) [inline templates/.snapshots/TestDatadogTf-monitoring-datadog.tf.tpl-monitoring-datadog.tf.snapshot:44]: Does this handle missing data? It kind of looks like you'll end up with a "No data" warning if this app doesn't exist. You probably need `default_zero` ... with `fill`? The docs seem to imply you can combine the two for a nonzero fill. :)

File: templates/monitoring/datadog.tf.tpl [hunk 3]
```diff
@@ -39,6 +39,38 @@ locals {
   ddTags = concat(["{{ .Config.Name }}", "team:{{  stencil.Arg "reportingTeam" }}"], var.additional_dd_tags)
 }
 
+resource "datadog_monitor" "argocd_application_health_status" {
+  type = "query alert"
+  name = "{{ .Config.Name | title }} argocd application health status"
+  query = "max({{ stencil.Arg "terraform.datadog.monitoring.argocd.appHealth.evaluationWindow" | default "last_15m"}}):clamp_max(sum:argocd_application_controller.argocd_app_info{name:{{ .Config.Name }},health_status:healthy} by {cluster_name}, 1) < 1"
+  tags = local.ddTags
+  message = <<EOF
+  If ArgoCD Health status is not healthy for a certain amount of time we want to know as it might indicated deployment issues.
```
Comment (jkinkead) [inline templates/monitoring/datadog.tf.tpl:48]: This should describe the state of the world this alert triggered on, like:
```suggestion
  ArgoCD Health status has been unhealthy over window {{ stencil.Arg "terraform.datadog.monitoring.argocd.appHealth.evaluationWindow" | default "last_15m"}}.
```

File: templates/monitoring/datadog.tf.tpl [hunk 4]
```diff
@@ -39,6 +39,38 @@ locals {
...
+  require_full_window = false
+}
+
+resource "datadog_monitor" "argocd_application_sync_status" {
+  type = "query alert"
+  name = "{{ .Config.Name | title }} argocd application sync status"
+  query = "max({{ stencil.Arg "terraform.datadog.monitoring.argocd.syncStatus.evaluationWindow" | default "last_15m"}}):clamp_max(sum:argocd_application_controller.argocd_app_info{name:{{ .Config.Name }},sync_status:synced} by {cluster_name}, 1) < 1"
+  tags = local.ddTags
+  message = <<EOF
+  If we ever have a pod restart, we want to know.
```
Comment (jkinkead) [inline templates/monitoring/datadog.tf.tpl:64]: Same comment: This should describe the state that triggered the alert, not why we have the alert.

**Review by jkinkead — COMMENTED** [review]
I don't think any of these queries handle no data gracefully.

No data should be green by default in most cases.

**Pull request merged**